### ISSUE TITLE
Question: Support for EnumDataType Annotation

### ISSUE THREAD

**Issue by JohnCatn** [RicoSuter/NJsonSchema#1323]
When using the EnumDataType Annotation to map an enum type to a string the enum type is being ignored.  Unfortunately I am modifying some legacy code and would like to be able to generate a schema which includes an enum for allowable values.

This can be done in [Json.Net ](https://www.newtonsoft.com/jsonschema/help/html/GenerateWithDataAnnotations.htm ) but I am preferring the additional attributed provided by NJsonSchema.

Have I missed a setting or is the EnumDataType attribute not yet supported?

Thanks

**Comment by RicoSuter** [comment]
Is this something which correctly works with Newtonsoft.Json but is not correctly picked up (same way) by NJS? So is NJS not behaving the same way as Newtonsoft.Json? If this is the case we should add it.

**Comment by RicoSuter** [comment]
We need to either fix it here: 
https://github.com/RicoSuter/NJsonSchema/blob/master/src/NJsonSchema/Generation/JsonSchemaGenerator.cs#L680

And/Or here: 
https://github.com/RicoSuter/NJsonSchema/blob/master/src/NJsonSchema/Generation/DefaultReflectionService.cs#L263

**Comment by JohnCatn** [comment]
Thanks for the pointer, I'll take a look.  Although I'm not sure my skills will be up to resolving it

**Comment by HIMISOCOOL** [comment]
Yeah this works flawlessly in Newtonsoft.Json.Schema but id love to use the MIT on this project. Can we get upated LOC @RicoSuter I might get some lockdown time to have a stab at this.
